frontend/src/metabase/query_builder/components/expressions/ExpressionEditorSuggestions/ExpressionEditorSuggestions.tsx

```diff
@@ -289,45 +305,42 @@ function ExpressionEditorSuggestionsListItem({
 function Footer({
   suggestion,
   highlightedIndex,
+  onHighlightSuggestion,
 }: {
   suggestion: WithIndex<SuggestionFooter>;
   highlightedIndex: number;
+  onHighlightSuggestion: (index: number) => void;
 }) {
   function handleMouseDownCapture(evt: MouseEvent) {
     // prevent the dropdown from closing
     evt.preventDefault();
   }
 
+  const handleMouseEnter = useCallback(() => {
+    onHighlightSuggestion(suggestion.index);
+  }, [suggestion.index, onHighlightSuggestion]);
+
+  const isHighlighted = highlightedIndex === suggestion.index;
+
   return (
     <ExpressionListFooter
       target="_blank"
       href={suggestion.href}
       onMouseDownCapture={handleMouseDownCapture}
-      isHighlighted={highlightedIndex === suggestion.index}
+      onMouseEnter={handleMouseEnter}
```

Comment:
Very minor but nonetheless an imperfection:

1. Hover an option (in the video it's "contains")
2. Press arrow down to select a different option
3. Move your mouse cursor within option highlighted in step 1

The option does not get highlighted

https://github.com/metabase/metabase/assets/6830683/4be5ac24-b4db-44a7-82df-b2d3895678df